text_editor.py

```diff
@@ -1,35 +1,90 @@
 """Text Editor"""
 
-# import keyboard
+# import curses
 
 class Model:
     """
     The model for the text editor.
     API supports up/down/left/right movement of cursor and insert/delete.
     TODO: Add highlighting.
+    TODO: Read/Write files.
     """
-    def __init__(self, cursor_position=0, prev_id=None,
-                 next_id=None, curr_id=0, iterator=1) -> None:
+
+    # window update constants:
+    def UP(self):
+        return "up"
+    def DOWN(self):
+        return "down"
+    def INSERT(self):
+        return "insert"
+    def DELETE(self):
+        return "delete"
+
+    def __init__(self, cursor_position=0, prev_id=None, next_id=None,
+                 curr_id=0, iterator=1, window=(10,16)) -> None:
         self.text = []
+        self.cache = {} # Stores the text, the previous id, and the next id of each line.
         self.cursor_position = cursor_position
 
-        self.prev_id = prev_id
-        self.next_id = next_id
-        self.curr_id = curr_id
-        self.iterator = iterator
+        # ids are keys for each line in the cache.
+        self.curr_id = curr_id # the id for the current line being manipulated.
+        self.prev_id = prev_id # the id for the previous line.
+        self.next_id = next_id # the id for the next line.
+
+        self.iterator = iterator # Creates monotonic identifiers so that new ids are guaranteed to be unique.
```

Comment:
The identifiers themselves aren't monotonic.
Maybe "Allocates monotonically increasing identifiers to guarantee new ids are unique."